Ginger/GingerCoreNET/RunLib/CLILib/CLIHelper.cs

```diff
@@ -236,34 +236,41 @@ private void SelectRunset()
         }
 
         private void SelectEnv()
-        {            
+        {
             Reporter.ToLog(eLogLevel.DEBUG, "Selected Environment: '" + Env + "'");
-            ProjEnvironment env = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Where(x => x.Name.ToLower().Trim() == Env.ToLower().Trim()).FirstOrDefault();
-            if (env != null)
-            {
-                mRunsetExecutor.RunsetExecutionEnvironment = env;
-            }
-            else
+            if (String.IsNullOrEmpty(Env) == false)
             {
-                if (Env == "Default")
+                ProjEnvironment env = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Where(x => x.Name.ToLower().Trim() == Env.ToLower().Trim()).FirstOrDefault();
+                if (env != null)
                 {
-                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().Count == 1)
-                    {
-                        mRunsetExecutor.RunsetExecutionEnvironment = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().First();
-                        Reporter.ToLog(eLogLevel.DEBUG, "Auto Selected Environment: '" + mRunsetExecutor.RunsetExecutionEnvironment.Name + "'");
-                    }
-                    else
+                    mRunsetExecutor.RunsetExecutionEnvironment = env;
+                    return;
+                }
+                else
+                {
+                    if (Env != "Default")
                     {
-                        Reporter.ToLog(eLogLevel.ERROR, "Cannot auto select default environment since solution do not contain 'Default' env and solution contains more than one env, please specify the env name using -e or --env");
+                        Reporter.ToLog(eLogLevel.ERROR, string.Format("Failed to find the Environment '{0}' in the Solution", Env));
+                        throw new Exception(string.Format("Failed to find the Environment '{0}' in the Solution", Env));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4812633)